newrelic/resource_newrelic_nrql_drop_rule_test.go

```diff
@@ -0,0 +1,100 @@
+// +build integration
+
+package newrelic
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccNewRelicNRQLDropRule_Basic(t *testing.T) {
+	rand := acctest.RandString(5)
+	description := fmt.Sprintf("nrql_drop_rule_%s", rand)
+	resourceName := "newrelic_nrql_drop_rule.foo"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckNewRelicNRQLDropRuleDestroy,
+		Steps: []resource.TestStep{
+			// Test: Create
+			{
+				Config: testAccNewRelicNRQLDropRuleConfig(description),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckNewRelicNRQLDropRuleExists(resourceName),
+				),
+			},
+			// Test: Import
+			{
+				ImportState:       true,
+				ImportStateVerify: true,
+				ResourceName:      resourceName,
+			},
+		},
+	})
+}
+
+func testAccCheckNewRelicNRQLDropRuleDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*ProviderConfig).NewClient
+	for _, r := range s.RootModule().Resources {
+		if r.Type != "newrelic_nrql_drop_rule" {
+			continue
+		}
+
+		accountID, ruleID, err := getNRQLDropRuleIDs(r.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		_, err = getNRQLDropRuleByID(client, accountID, ruleID)
+
+		if err == nil {
+			return fmt.Errorf("drop rule still exists: %s", err)
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckNewRelicNRQLDropRuleExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("not found: %s", n)
+		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no ID is set")
+		}
+
+		client := testAccProvider.Meta().(*ProviderConfig).NewClient
+
+		accountID, _, err := getNRQLDropRuleIDs(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		_, err = client.Nrqldroprules.GetList(accountID)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func testAccNewRelicNRQLDropRuleConfig(description string) string {
```

Comment:
It would be good to have a test for various actions, and for `account_id` inheritance.